hls4ml/backends/vivado/vivado_backend.py

```diff
@@ -376,37 +373,14 @@ def init_depconv2d(self, layer):
         )  # TODO Once we have SeparableConv implementation for io_parallel this should be set properly
         layer.set_attr('implementation', layer.model.config.get_conv_implementation(layer).lower())
 
-    def _set_pooling_accum_t(self, layer, pool_size):
```

Comment:
We shouldn't merge this change before we have the subsequent change in auto precision inference. Right now it's removing features that already exist (and catapult customers may be using) without replacing it right away. For Vitis and Vivado it is fine to not add something that doesn't yet exist, but one has to be more careful about removing already existing features. If we want to decouple things, we should move the catapult update to a separate PR.

Reply:
The current `_set_pooling_accum_t` is flawed (overrides `default_t` in most cases, doesn't always fix overflow as well; multiple pooling layers will results in a huge default model bitwidth thanks to the `+=` operation), and I fixed it in the last iteration of this PR.

This PR will not work without proper `accum_t` set, and that's why I fixed it right here in the last iteration. As you suggested, I integrated it with #855, and I have marked #855 as required by this PR.

Reply:
Generally, unless they are code maintenance-type changes or purely adding new features, Catapult backend changes need Siemens approval.

Reply:
We are hoping to merge #855 soon. The last changes failed the pytests, so after chaging "ave" to "average" we are trying again. Vivado should not have the overrides since it never had them. It should start with the new way of doing things from the beginning.

Reply:
I would probably make the Catapult change a separate PR, though. We do eventually want that.

Reply:
As #855 does not remove that override, I will remove the Vivado override and leave the Siemens one as-is.

Reply:
I hadn't appreciated that there was already some auto precision setting for Vivado. Nevertheless, I believe removing it is the right thing to do since we have the auto precision. I will announce it.